addons/project/tests/test_project_sharing.py

```diff
@@ -85,24 +84,162 @@ def test_project_share_wizard(self):
+        self.assertEqual(len(project_share_wizard.collaborator_ids), 2, 'The access right added in project share wizard should be added in the project when the user confirm the access in the wizard.')
+        for collaborator in project_share_wizard.collaborator_ids:
+            collaborator_vals = {
+                'partner_id': collaborator.partner_id,
+                'access_mode': collaborator.access_mode,
+            }
+            if collaborator.partner_id == self.user_portal.partner_id:
+                self.assertDictEqual(collaborator_vals, {
+                    'partner_id': self.user_portal.partner_id,
+                    'access_mode': 'edit_limited',
+                })
+            else:
+                self.assertDictEqual(collaborator_vals, {
+                    'partner_id': self.partner_1,
+                    'access_mode': 'edit',
+                })
+
+    def test_project_share_wizard_remove_collaborators(self):
+        PortalShare = self.env['project.share.wizard'].with_context(active_model="project.project", active_id=self.project_portal.id)
+        self.project_portal.write({
+            'collaborator_ids': [
+                Command.create({'partner_id': self.user_portal.partner_id.id}),
+                Command.create({'partner_id': self.partner_1.id, 'limited_access': True}),
+            ],
+        })
+        self.project_portal.message_subscribe(partner_ids=[self.partner_2.id])
+        with Form(PortalShare) as project_share_form:
+            self.assertEqual(len(project_share_form.collaborator_ids), 3, "2 external collaborators should be found for that project.")
+            collaborator_vals_per_id = project_share_form.collaborator_ids._field_value._data
+            collaborator_access_mode_per_partner_id = {
+                c['partner_id']: c['access_mode']
+                for c in collaborator_vals_per_id.values()
+            }
+            self.assertIn(self.user_portal.partner_id.id, collaborator_access_mode_per_partner_id)
+            self.assertIn(self.partner_1.id, collaborator_access_mode_per_partner_id)
+            self.assertIn(self.partner_2.id, collaborator_access_mode_per_partner_id)
+            access_mode_expected_per_partner_id = {
+                self.user_portal.partner_id.id: 'edit',
+                self.partner_1.id: 'edit_limited',
+                self.partner_2.id: 'read',
+            }
+            self.assertDictEqual(collaborator_access_mode_per_partner_id, access_mode_expected_per_partner_id)
+            collaborator_ids_to_remove = {c_id for c_id, vals in collaborator_vals_per_id.items() if vals['access_mode'] != 'read'}
+            index = 0
+            for collaborator_id in project_share_form.collaborator_ids.ids:
+                if collaborator_id in collaborator_ids_to_remove:
+                    project_share_form.collaborator_ids.remove(index)
+                else:
+                    index += 1
```

Comment:
Nope, because `collaborator_ids` is not really a recordset is like a O2MProxy since `project_share_form` is `Form` and so we don't have any filtered method. 😕